datafusion/core/src/execution/context/csv.rs

```diff
@@ -116,11 +116,11 @@ mod tests {
 
         assert_eq!(results.len(), 1);
         let expected = [
-            "+--------------+--------------+----------+",
-            "| sum(test.c1) | sum(test.c2) | count(*) |",
-            "+--------------+--------------+----------+",
-            "| 10           | 110          | 20       |",
-            "+--------------+--------------+----------+",
+            "+--------------+--------------+--------------+",
+            "| sum(test.c1) | sum(test.c2) | count_star() |",
```

Comment:
It would be great if this could stay `count(*)`

Perhaps that is why you implemented `AggregateUDFImpl::display_name` 🤔 

- https://github.com/apache/datafusion/pull/14695

Reply:
Actually #14695 is for others to keep `count(*)` given that I prefer `count_star()` :)

Reply:
FWIW, DuckDB formats `COUNT(*)` as `count_star` and preserves `COUNT(<expr>)` where `expr` isn't a wildcard.

In this case, it looks like we don't preserve `COUNT(1)`.

Reply:
> In this case, it looks like we don't preserve COUNT(1).

Yes we don't. count(const_expr), count(), count(*) are all the same thing, I don't think we need to preserve them all

Reply:
> Yes we don't. count(const_expr), count(), count(*) are all the same thing, I don't think we need to preserve them all

fine for me.

calling it just `count()` (or `count(1)`) would be shorter and cleaner


in particular, the `count_star` function should not exist as user-callable syntax, (`SELECT count_star()` should fail with "function not found")

Reply:
> It would be great if this could stay `count(*)`

i agree ...

> Perhaps that is why you implemented `AggregateUDFImpl::display_name` 🤔


... but not at any cost.
if displaying `count(*)` rather than (equivalent) `count(1)` requires ~200 lines of duplicated code, i'd stick with `count(1)`